### PR TITLE
Fix: Add support for index and series to utils.to_filewise_index()

### PR DESCRIPTION
**Pull request by frankenjoe**
Closes https://github.com/audeering/audformat/issues/281

Also covers the case that the input is empty.

### CODE REVIEW

**Comment by frankenjoe** [comment]
I cleaned up the test and added cases with series / index as input, which are now failing as expected.

**Comment by hagenw** [comment]
Great, could you maybe also add an example to the docstring?

**Comment by frankenjoe** [comment]
> Great, could you maybe also add an example to the docstring?

It's not so easy as In contrast to `to_segmented_index()` the files have to exist.

File: tests/test_utils.py [hunk 1]
```diff
@@ -2571,75 +2571,133 @@ def test_to_segmented_index(obj, allow_nat, files_duration, root, expected):
...
+            audformat.segmented_index(),
+            [],
+        ),
+        (
+            pd.Series(index=audformat.filewise_index(), dtype='object'),
+            [],
+        ),
+        (
+            pd.Series(index=audformat.segmented_index(), dtype='object'),
+            [],
```
Comment (hagenw) [inline tests/test_utils.py:2591]:
```suggestion
        (
            pd.Series(index=audformat.filewise_index(), dtype='object'),
            [],
        ),
        (
            pd.Series(index=audformat.segmented_index(), dtype='object'),
            [],
        ),
        (
            pd.DataFrame(index=audformat.filewise_index(), dtype='object'),
            [],
        ),
        (
            pd.DataFrame(index=audformat.segmented_index(), dtype='object'),
            [],
```

File: tests/test_utils.py [hunk 2]
```diff
@@ -2571,75 +2571,133 @@ def test_to_segmented_index(obj, allow_nat, files_duration, root, expected):
...
         (
-            'tmp',
-            'segments',
+            pytest.DB['segments'].get(),
             [
-                str(i).zfill(3) + f'_{j}'
-                for i in range(1, 11)
-                for j in range(10)
-            ]
+                str(i + 1).zfill(3) + f'_{j}'
```
Comment (hagenw) [inline tests/test_utils.py:2597]: Could you maybe add a comment above this statement what this generates as I have a hard time understanding it, e.g.

```python
# f1_0, f1_1, ...
```

Comment (frankenjoe) [reply tests/test_utils.py]: I had copied it from the previous test. I now switched to `f'...'` notation and also added a comment.

Comment (hagenw) [reply tests/test_utils.py]: Thanks.

**Comment by hagenw** [comment]
> > Great, could you maybe also add an example to the docstring?
> 
> It's not so easy as In contrast to `to_segmented_index()` the files have to exist.

OK, let's not make this part of this pull request.

**Pull request merged**